impl/src/main/java/org/jboss/cdi/tck/tests/extensions/configurators/BeanAttributesConfigurator/BeanAttributesConfiguratorTest.java

```diff
@@ -0,0 +1,112 @@
+import org.jboss.cdi.tck.AbstractTest;
+import org.jboss.cdi.tck.shrinkwrap.WebArchiveBuilder;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.test.audit.annotations.SpecAssertion;
+import org.jboss.test.audit.annotations.SpecAssertions;
+import org.jboss.test.audit.annotations.SpecVersion;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author <a href="mailto:manovotn@redhat.com">Matej Novotny</a>
+ */
+@Test
+@SpecVersion(spec = "cdi", version = "2.0-EDR1")
+public class BeanAttributesConfiguratorTest extends AbstractTest {
+
+    public static final String SWORD_NAME = "Frostmourne";
+
+    @Deployment
+    public static WebArchive createTestArchive() {
+        return new WebArchiveBuilder().withTestClassPackage(BeanAttributesConfiguratorTest.class)
+            .withExtension(ProcessBeanAttributesObserver.class).build();
+    }
+
+    @Test
+    @SpecAssertions({ 
+        @SpecAssertion(section = BEAN_ATTRIBUTES_CONFIGURATOR, id = "aa"),
+        @SpecAssertion(section = BEAN_ATTRIBUTES_CONFIGURATOR, id = "ba"),
+        @SpecAssertion(section = BEAN_ATTRIBUTES_CONFIGURATOR, id = "bd"),
+        @SpecAssertion(section = BEAN_ATTRIBUTES_CONFIGURATOR, id = "bf"),
+        @SpecAssertion(section = BEAN_ATTRIBUTES_CONFIGURATOR, id = "bh") })
+    public void testSingleAdditionMethods() {
+        Bean<Sword> bean = getUniqueBean(Sword.class, new TwoHanded.TwoHandedLiteral());
+
+        assertTrue(bean.getQualifiers().contains(new TwoHanded.TwoHandedLiteral()));
+        assertTrue(bean.getName().equals(SWORD_NAME));
+        assertTrue(bean.getTypes().contains(Weapon.class));
+        assertTrue(bean.getStereotypes().contains(Equipment.class));
+    }
+
+    @Test
+    @SpecAssertions({ 
+        @SpecAssertion(section = BEAN_ATTRIBUTES_CONFIGURATOR, id = "ab"),
+        @SpecAssertion(section = BEAN_ATTRIBUTES_CONFIGURATOR, id = "bb"),
+        @SpecAssertion(section = BEAN_ATTRIBUTES_CONFIGURATOR, id = "bc"),
+        @SpecAssertion(section = BEAN_ATTRIBUTES_CONFIGURATOR, id = "be"),
+        @SpecAssertion(section = BEAN_ATTRIBUTES_CONFIGURATOR, id = "bg"),
+        @SpecAssertion(section = BEAN_ATTRIBUTES_CONFIGURATOR, id = "bi") })
+    public void testMultiAdditionMethods() {
+        Bean<Axe> bean = getUniqueBean(Axe.class, new TwoHanded.TwoHandedLiteral(), new Reforged.ReforgedLiteral());
+        
+        Set<Annotation> qualifiers = bean.getQualifiers();
+        Set<Class<? extends Annotation>> stereotypes = bean.getStereotypes();
+        Set<Type> types = bean.getTypes();
+        
+        assertTrue(bean.getScope().equals(RequestScoped.class));
+        assertTrue(qualifiers.contains(new Reforged.ReforgedLiteral()) && qualifiers.contains(new TwoHanded.TwoHandedLiteral()));
+        assertTrue(stereotypes.contains(Melee.class) && stereotypes.contains(Equipment.class));
+        assertTrue(types.contains(Weapon.class) && types.contains(Tool.class));
```

Comment:
I would replace these logical ands. It would be better to compare entire collection.